### ISSUE TITLE
Clear icon is not accessible

### ISSUE THREAD

**Issue by danwellman** [ng-select/ng-select#1895]
**Describe the bug**
The clear icon that is added to the select after making a selection does not have `tabindex` and cannot be reached with the keyboard. Additionally, the `aria-hidden` attribute is set to true while the clear icon is visible, which seems wrong and misleading for users of assistive technology

**Reproducible example**
Any of the examples on ng-select.github.io, e.g. https://ng-select.github.io/ng-select#/data-sources

**To Reproduce** 
Steps to reproduce the behavior (if example is not provided):
1. Go to any of the ng-select examples
2. Try to use the keyboard to clear a selection
3. Fail

**Expected behavior**
The clear icon should be fully keyboard accessible and display appropriate aria states

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 90.0.4430.93
